src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Figure newFigure = null;
+        Color color = ColorSupplier.getRandomColor();
+        int figureIndex = new Random().nextInt(FigureTypes.values().length);
+        FigureTypes newFigureType = FigureTypes.values()[figureIndex];
+        switch (newFigureType) {
+            case CIRCLE:
+                int radius = new Random().nextInt(20);
+                newFigure = new Circle(color, radius);
+                break;
```

Comment:
create a private method that will create the shape and call it here
```suggestion
                return getRandomCircle();
```